controllers/util/provider.go

```diff
@@ -77,3 +79,8 @@ func readAndDecodePrivateKey(privateKeyPath string) (*rsa.PrivateKey, error) {
 	}
 	return privateKey, nil
 }
+
+func GetMD5Hash(text string) string {
+	hash := md5.Sum([]byte(text))
+	return hex.EncodeToString(hash[:])
+}
```

Comment:
This is not a provider-specific util, it is just an MD5 hashing function. Please move it to `util.go`